include/envoy/http/filter.h

```diff
@@ -28,9 +28,9 @@ namespace Http {
 enum class FilterHeadersStatus {
   // Continue filter chain iteration.
   Continue,
-  // Do not iterate to any of the remaining filters in the chain. Returning
-  // FilterDataStatus::Continue from decodeData()/encodeData() or calling
-  // continueDecoding()/continueEncoding() MUST be called if continued filter iteration is desired.
+  // Do not iterate to any of the remaining filters in the chain.
+  // continueDecoding() or continueEncoding() MUST be called if continued filter iteration is
```

Comment:
This is not true, StopIteration on headers doesn't stop onData being called, so in the same filter you can return `Continue` without calling continueDecoding()/Encoding() to continue.

Reply:
Really?

So if I want to send a local reply from decodeHeaders() the only option to halt everything would be StopAllIterationAndWatermark? Because StopAllIterationAndBuffer would make less sense.

Otherwise, my filter would have to keep track of the local reply so that it would return early from decodeData().

Reply:
My actual aim is to add a comment to clarify what's the right value after issuing a `sendLocalReply()` or `decoder_callbacks_->encodeHeaders(std::move(response_headers), true /* end_stream */, ...);` from `decodeHeaders()`.

Reply:
I've been kind of curious about this, most cases are set up so that header status can return `StopIteration` after a send local reply, and then be able to control how data is handled in `decodeData()` (maybe buffered by envoy, buffered in the filter, stopped, continue).

So maybe you need to keep track of the local reply state and then return `StopIterationNoBuffer` from `decodeData()` in that state if you don't care about if data being buffered.

Reply:
Ok, if this is the case let's document it (here and somewhere else). I'll wait for Lizan to chime in and then I'll update the PR. 

It's still weird (to me) that you'd send a local response and still care about what happens next, it should be game over at that point (specially because other filters might not know that a local response was already emitted and do something buggy -- no?).

Reply:
Update: I think the comment was correct originally, and that when sendLocalReply is sent, the filter manager will stop decoding data? https://github.com/envoyproxy/envoy/blob/0dd4b617fbf3578c496750ec91dffd7308028a04/source/common/http/filter_manager.cc#L503.

I hope the same holds true with StopAllIterations...

Reply:
Yeah, once a stream is ended via local reply, the filter manager will stop raising events, so this is correct. cc @snowp @alyssawilk

Reply:
How about this https://github.com/envoyproxy/envoy/pull/13678/commits/032fddf8fc85d99242e1b7de9fa180802ef4e3ff?